agent/structs/structs.go

```diff
@@ -996,7 +1000,8 @@ type NodeService struct {
 	Address           string
 	TaggedAddresses   map[string]ServiceAddress `json:",omitempty"`
 	Meta              map[string]string
-	Port              int
+	Port              int    `json:",omitempty"`
+	SocketPath        string `json:",omitempty"` // This might be integrated into Address somehow, but not sure about the ergonomics. Only one of (address,port) or socketpath can be defined.
```

Comment:
Did you intend to keep this comment here? If so maybe it needs a TODO annotation.